File: tests/positives/framework-detection/go/detect-openai.go
package main

// Positive test cases for OpenAI (Go)

// ruleid: detect-openai
import "github.com/sashabaranov/go-openai"

func main() {
	// ruleid: detect-openai
	client := openai.NewClient("sk-test")

	// ruleid: detect-openai
	resp, _ := client.CreateChatCompletion(ctx, openai.ChatCompletionRequest{
		Model: openai.GPT4,
	})

	// ruleid: detect-openai
	client.CreateEmbeddings(ctx, openai.EmbeddingRequest{})
}

File: tests/positives/framework-detection/csharp/detect-openai.cs
// Positive test cases for OpenAI (C#)

// ruleid: detect-openai
using OpenAI;

// ruleid: detect-openai
using OpenAI.Chat;

public class Example
{
    public async Task Run()
    {
        // ruleid: detect-openai
        var client = new OpenAIClient("sk-test");

        // ruleid: detect-openai
        ChatClient chatClient = new ChatClient("gpt-4", "sk-test");

        // ruleid: detect-openai
        var completion = await chatClient.CompleteChatAsync("Hello");

        // ruleid: detect-openai
        var stream = chatClient.CompleteChatStreamingAsync("Stream test");
    }
}
